cluster/core/src/main/resources/reference.conf

```diff
@@ -15,9 +15,12 @@ lagom.cluster {
   join-self = ${lagom.defaults.cluster.join-self}
 
   # Exit the JVM forcefully when the ActorSystem has been terminated.
-  # This is by default off, but you may want to turn it on in production
-  # to restart the process.
-  exit-jvm-when-system-terminated = off
+  exit-jvm-when-system-terminated = on
```

Comment:
I don't think you want this on in development mode. It would terminate the entire sbt process.

Reply:
Correct, in that case, we must switch to `off` in dev and in test.

Reply:
@TimMoore, I set it to `off` in Dev and Test. 

We have been through a similar situation recently with the introduction of `lagom.cluster.bootstrap.enabled`. Enabled by default, but disable in Dev and Test. I just added `lagom.cluster.exit-jvm-when-system-terminated = off` next to it.

Alternatively, we could have just adapted the `JoinClusterImpl` and only exit the jvm if the property is `on` and `environment.mode == Mode.Prod`.

That feature really only makes sense on Prod environment and the user can still disable it if they have some special needs. 

WDYT?

(cc @marcospereira, @dwijnand, @ignasi35)

Reply:
I like the trend to make the Lagom default settings be production-aware.

There is an inconsistent approach to default settings but tending to production-first and use overrides in test and dev modes is the right move. +1 !